terraform/files/bin/upload_capi_image.sh

```diff
@@ -4,19 +4,30 @@
 
 ~/bin/fixup_k8s_version.sh $CCCFG
 KUBERNETES_VERSION=$(yq eval '.KUBERNETES_VERSION' $CCCFG)
+. ~/bin/parse_k8s_version.inc
 #UBU_IMG_NM=ubuntu-capi-image-$KUBERNETES_VERSION
 UBU_IMG_NM=$(yq eval '.OPENSTACK_IMAGE_NAME' $CCCFG)
```

Comment:
I see one possible edge case that we could consider:

`OPENSTACK_IMAGE_NAME` variable consists of a static string and `kubernetes_version`  variable, i.e. `ubuntu-capi-image-${kubernetes_version}`

IIUC the OpenStack image name for both supported Ubuntu versions (20.04 and 22.04) will be the same if the k8s version will not change e.g. for the default k8s v1.25.11 the OpenStack image name for both Ubuntu versions will be  `ubuntu-capi-image-v1.25.11`. 

IMO This may cause an unwanted behavior when the user OpenStack project already contains Ubuntu 20.04 image for the k8s version v1.25.11 or newer. Then the Ubuntu 22.04 image won't be uploaded to the user's OpenStack project as the image with the desired name is already there, see [upload_capi_image.sh#L20](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/files/bin/upload_capi_image.sh#L20)

Possible solutions:
1. Document the above and force users to remove "conflicting" images from their OS projects 
2. Introduce the Ubuntu version in  `OPENSTACK_IMAGE_NAME` variable, i.e. `ubuntu-${ubuntu_version}-capi-image-${kubernetes_version}`  (maybe consider applying this only for Ubuntu 22.04 and newer to avoid unwanted uploads of already uploaded 20.04 images)  (IMO: this looks good to me)
3. Replace OpenStack image if Ubuntu 22.04 and  k8s version v1.25.11 or newer are used

Reply:
Hi, @chess-knight, as discussed with @matofeder it is enough to do just document the change but as there's nothing expected to break users should not be forced to remove 20.04er images since they will get the 22.04 automatically when they update regularly.